src/ints/mouse.cpp

```diff
@@ -31,21 +33,47 @@
 #include "bios.h"
 #include "dos_inc.h"
 
+enum DOS_EV:uint8_t { // compatible with DOS driver mask in driver function 0x0c
+    NOT_DOS_EVENT   = 0x00,
+    MOUSE_HAS_MOVED = 0x01,
+    PRESSED_LEFT    = 0x02,
+    RELEASED_LEFT   = 0x04,
+    PRESSED_RIGHT   = 0x08,
+    RELEASED_RIGHT  = 0x10,
+    PRESSED_MIDDLE  = 0x20,
+    RELEASED_MIDDLE = 0x40,
+    WHEEL_HAS_MOVED = 0x80,
+};
+
+static const uint8_t QUEUE_SIZE  = 32;   // if over 255, increase 'queue_used' type size
+static const uint8_t KEY_MASKS[] = { 0x01, 0x02, 0x04, 0x08, 0x10 };
+
+static uint8_t buttons_12  = 0; // state of buttons 1 (left), 2 (right), as visible on host side
+static uint8_t buttons_345 = 0; // state of mouse buttons 3 (middle), 4, and 5 as visible on host side
+
+typedef struct MouseEvent {
+    uint8_t dos_type    = 0;
+    uint8_t dos_buttons = 0; 
+
+    MouseEvent() {}
+    MouseEvent(uint8_t dos_type) : dos_type(dos_type) {}
+
+} MouseEvent;
```

Comment:
We had a chat about this with @kcgen recently, and the way forward seems to be CamelCase for enum values and underscore_case for constants.